frontend/src/components/HeatMap/HeatMap.js

```diff
@@ -1,6 +1,73 @@
 import * as d3 from 'd3';
 
-export const drawFlatHeatMap = () => 0;
+
+/**
+ * Flatten data when it comes in and draw heat map visualization for rubric performance.
+ * @param {string} id ID of the div to target to draw the vis.
+ * @param {array} data The datapoints to iterate over for visualization.
+ * @returns {none} Produces side-effect which draws heat map.
+ * data = [
+ *   [
+ *     {criterion: "Bibliography", ratingDescription: "Outstanding", value: 10},
+ *     {criterion: "Bibliography", ratingDescription: "Average", value: 20},
+ *     {criterion: "Bibliography", ratingDescription: "Poor", value: 70}
+ *   ],
+ *   [
+ *     {...},{...},{...}
+ *   ]
+ * ]
+ */
```

Comment:
It might be useful to more explicitly document the assumption on the incoming data for the heatmap visualization and the expected transformation. For example, you could  extend the description to include something like:

```
It is the caller's responsibility to ensure that each criterion in the rubric 
contains the same set of rating descriptions. 

For example, this nested data:

data = [
  [
    {criterion: "Bibliography",ratingDescription: "Outstanding", value: 10},
    {criterion: "Bibliography",ratingDescription: "Average", value: 20},
    {criterion: "Bibliography",ratingDescription: "Poor", value: 70}
  ],
  [
    {criterion: "Closing",ratingDescription: "Outstanding", value: 70},
    {criterion: "Closing",ratingDescription: "Average", value: 20},
    {criterion: "Closing",ratingDescription: "Poor", value: 70}
  ],
  [
    {...}, {...}, {...}
  ]
]

Will be flattened to:

[
  {"criterion": "Bibliography", "ratingDescription": "Outstanding", "value": 10},
  {"criterion": "Bibliography","ratingDescription": "Average", "value": 20},
  {"criterion": "Bibliography", "ratingDescription": "Poor", "value": 70},
  {"criterion": "Closing", "ratingDescription": "Outstanding", "value": 70},
  {"criterion": "Closing", "ratingDescription": "Average", "value": 20},
  {"criterion": "Closing", "ratingDescription": "Poor", "value": 70},
  {...},
  {...},
  {...}
]
```

Reply:
Good idea. I'll update it.

Reply:
Having such a large comment required for a function makes me think there is something wrong with the function. Too many assumptions?